src/operator/numpy/random/np_bernoulli_op.h

```diff
@@ -0,0 +1,206 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * Copyright (c) 2019 by Contributors
+ * \file np_bernoulli_op.h
+ * \brief Operator for numpy sampling from bernoulli distribution.
+ */
+#ifndef MXNET_OPERATOR_NUMPY_RANDOM_NP_BERNOULLI_OP_H_
+#define MXNET_OPERATOR_NUMPY_RANDOM_NP_BERNOULLI_OP_H_
+
+#include <mxnet/operator_util.h>
+#include <algorithm>
+#include <string>
+#include <vector>
+#include "../../elemwise_op_common.h"
+#include "../../mshadow_op.h"
+#include "../../mxnet_op.h"
+#include "../../operator_common.h"
+#include "../../tensor/elemwise_binary_broadcast_op.h"
+#include "./dist_common.h"
+
+namespace mxnet {
+namespace op {
+
+struct NumpyBernoulliParam : public dmlc::Parameter<NumpyBernoulliParam> {
+  dmlc::optional<float> prob;
+  dmlc::optional<float> logit;
+  std::string ctx;
+  int dtype;
+  bool is_logit;
+  dmlc::optional<mxnet::Tuple<int>> size;
```

Comment:
Why should we use `Tuple<int>`? Should we use `TShape`?

Reply:
They are, in essence, the same thing I believe.
I actually copy the declaration from
https://github.com/apache/incubator-mxnet/blob/a783d8119260fa370c8b12d3154ebc27e8e7284d/src/operator/numpy/random/np_multinomial_op.h#L53
As multinomial is the first re-implemented random OP in Deep Numpy,  random operators after multinomial all declare size as optional<mxnet::Tuple<int>>.